test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -433,8 +448,12 @@ func TestTierTemplateRevision(t *testing.T) {
 		assert.LessOrEqual(t, len(objs.Items), 6)
 		// we check that the TTR content has the parameters replaced with values from the NSTemplateTier
 		for _, obj := range objs.Items {
-			assert.Contains(t, string(obj.Spec.TemplateObjects[0].Raw), "janedoe")        // the object should have the namespace name replaced
-			assert.NotContains(t, string(obj.Spec.TemplateObjects[0].Raw), ".SPACE_NAME") // the object should NOT contain the variable anymore
+			// the object should have all the variables still there since this one will be replaced when provisioning the Space
+			assert.Contains(t, string(obj.Spec.TemplateObjects[0].Raw), ".SPACE_NAME")
+			assert.Contains(t, string(obj.Spec.TemplateObjects[0].Raw), ".DEPLOYMENT_QUOTA")
+			// the parameter is copied from the NSTemplateTier
+			assert.Equal(t, obj.Spec.Parameters[0].Name, customTier.Spec.Parameters[0].Name)
```

Comment:
If I understand the logic correctly, both parameters should now be copied from the NSTemplateTier, right?

Reply:
There is a distinction between "static" and "dynamic" parameters:

1. **static parameters** are those defined in the `NSTemplateTier.Spec.Parameters` field ( like the `DEPLOYMENT_QUOTA` ) and they will be copied to the `TierTemplateRevision.Spec.Parameters` 
2. **dynamic parameters** are not defined in `NSTemplateTier.Spec.Parameters`, they are just referenced in the `TierTemplate.Spec.TemplateObjects` and in the `TierTemplateRevision.Spec.TemplateObjects` ( as of now we only have the `SPACE_NAME` ) , and they will be evaluated _dynamicallly_ by the nstemplateset_controller when provisioning the namespace for a user

This is why the test only expects one parameter even if there are two variables in the `TemplateObjects`. I can maybe add a comment there explaining.

Reply:
added a comment in https://github.com/codeready-toolchain/toolchain-e2e/pull/1061/commits/f818ea94cc6055e1bd73543a0edfa2e6263e8469